crates/bevy_reflect/src/func/args/from_arg.rs

```diff
@@ -0,0 +1,90 @@
+use crate::func::args::{Arg, ArgError, ArgInfo};
+
+/// A trait for types that can be created from an [`Arg`].
+pub trait FromArg {
+    /// The type of the item created from the argument.
+    ///
+    /// This should almost always be the same as `Self`, but with the lifetime `'a`.
+    type Item<'a>;
```

Comment:
This is what people who opposed GATs were warning us about 😨

Reply:
😈